### DISCUSSION TITLE
Public API

### DISCUSSION THREAD

**Discussion by sirdir1972** [tolwi/hassio-ecoflow-cloud#344] [General]
Hi!
I’ve been using the integration for a long time, but the private api doesn’t seem to work very well anymore. 
I can’t find anything about how to use the public api… Can someone point me to the right direction?

**Comment by dltccf** [comment]
To use the public API you have to register at developer.ecoflow.com and then wait a week for them to make you a developer. Then you can generate an accesskey.

**Comment by bluesk1d** [comment]
I've gone through this process and am getting an "accessKey is invalid" error on setup. I'm quite sure I am copying and pasting correctly from my profile at developer.ecoflow.com. Anyone else getting that?

Edit: Found this [post](https://www.reddit.com/r/Ecoflow_community/comments/1fcfdl6/comment/lr4r9st/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button) which fixed the accessKey issue. Now it returns "signature is wrong".

**Reply by pgee70** [reply]
did you make sure you are entering the correct API URL? - there seem to be two - api-a.ecoflow.com and api-e.ecoflow.com 
e is for europe, a is for america. i am in Australia and use the api-a domain name. if you live outside those areas - take a punt on the other one.  i looked at the ecoflow app code structure on ios and there was some country - to - api mapping there.

i am using the public ecoflow api via node-red plugin with home assistant on a raspberry pi4
i installed a plugin/palette @shaoranlaos/node-red-contrib-ecoflow-http-api for node-red and entered the server/access key and secret key/ device serial and i was able to view the device data and programmatically change it with node-red.
there seems to be limited functionality with the api - but i do have it doing what i need it to.

**Reply by bluesk1d** [reply]
Thanks for the info. Yeah that's the handy tidbit that got me past the invalid accesskey issue. I'm not sure about the signature error though. I don't currently use node red.

**Reply by pgee70** [reply]
if you read the developer documentation on https://developer.ecoflow.com/us/document/generalInfo , you can see that it takes quite a bit of processing to convert/encrypt your developer/access key into a signing key.  i didn't want to do that, so used the available libraries.

**Comment by getastro71** [comment]
Hello all, so I am trying to set up  my ecoflow delta pro.  I have  everything installed and changed the api address, but when I but in my accesskey and secretkey I get back unknow and a close button.  What would cause that?

**Reply by pgee70** [reply]
you need to include a step-by-step explanation of what you are doing.
'when i but in my access key and secret key'. where do you input these things?
explain what you are doing, where you are doing it, what you expected, and what you observed.
what platform are you using? 
what are you entering the accesskey and secret key into? 
what model of delta pro are you using?
how did you calculate/create these data?

**Reply by koolin** [reply]
Running into this myself.  The unknown seems related to the retrieval of devices.

![{5C6C1A3F-EAA5-4C6E-9AF8-8895348C63A2}](https://github.com/user-attachments/assets/80897773-f099-43a9-afb4-26e3464d9599)

from config_flow.py

`try:
                devices = await self.auth.fetch_all_available_devices()
                self.set_device_list(devices)
            except Exception:  # pylint: disable=broad-except
                _LOGGER.exception("Unexpected exception in fetch device action")
                return self.async_abort(reason="unknown")`

Not sure what the solution is.  I have the devices registered against my ecoflow account.
